elastic.py

```diff
@@ -1,71 +1,83 @@
-        self.config = elasticConfig
+    def __init__(self, autonPlans, activeStartingPositions):
         self.autonPlans = autonPlans
-        self.autonConfig = autonConfig
-        self.selectedTaskKey = autonConfig['B_THREE_NOTE_AMP_SIDE']
-        self.isNoteDetected = self.config["NOTE_IS_DETECTED"]
-        self.isNoteLeft = self.config["NOTE_ON_LEFT"]
-        self.isNoteRight = self.config["NOTE_ON_RIGHT"]
-        self.isNoteLoaded = self.config["NOTE_IS_LOADED"]
-
-    def displayMainWindow(self):
-        tab = "Main Window"
-
-        (Shuffleboard.getTab(tab)
-        .add("Note Detected!", self.isNoteDetected)
-        .withSize(4,4)
-        .withPosition(17,0)
-        .withWidget(BuiltInWidgets.kBooleanBox)
-        .getEntry())
-
-        (Shuffleboard.getTab(tab)
-        .add("Note on Left", self.isNoteLeft)
-        .withSize(4,4)
-        .withPosition(13,0)
-        .withWidget(BuiltInWidgets.kBooleanBox)
-        .getEntry())
-
-        (Shuffleboard.getTab(tab)
-        .add("Note on Right", self.isNoteRight)
-        .withSize(4,4)
-        .withPosition(21,0)
-        .withWidget(BuiltInWidgets.kBooleanBox)
-        .getEntry())
-
-        (Shuffleboard.getTab(tab)
-        .add("Note Loaded!",self.isNoteLoaded)
-        .withSize(4,4)
-        .withPosition(17,4)
-        .withWidget(BuiltInWidgets.kBooleanBox)
-        .getEntry())
+        self.startingPositions = activeStartingPositions
+        self.isElasticSubmitted = False
+        self.controllerDriverElastic = False
+        self.controllerOperatorElastic = False
+        self.beamBreakBoolean = False
+    
+    def teamDisplay(self, defaultColor, defaultLabel):
+        self.teamChooser = wpilib.SendableChooser()
+        color = (defaultColor + " Alliance")
+        label = defaultLabel
+        self.teamChooser.setDefaultOption(color, label)
+        if defaultColor == 'Red':
+            self.teamChooser.addOption("Blue Alliance", "BLUE")
+        else:
+            self.teamChooser.addOption("Red Alliance", "RED")
+        SmartDashboard.putData(self.teamChooser)
+    
+    def positionDisplay(self, defaultPosition):
+        self.positionChooser = wpilib.SendableChooser() 
+        position = ("Positon " + defaultPosition)
+        defaultLabel = defaultPosition
+        self.positionChooser.setDefaultOption(position, defaultLabel)
+        for i in self.startingPositions:
+            position = ("Positon " + i)
+            label = i
+            if not(defaultLabel == label):
+                self.positionChooser.addOption(position, label)
+        SmartDashboard.putData(self.positionChooser)
 
     def autonDisplay(self):
         self.chooser = wpilib.SendableChooser()
         self.chooser.setDefaultOption(self.autonPlans[0], self.autonPlans[0])
-
         for plan in self.autonPlans:
             self.chooser.addOption(plan, plan)
-
         SmartDashboard.putData(self.chooser)
+    
+    def updateBeamDisplay(self, noteDetected):
```

Comment:
Could this be simplified to only be `self.putBoolean("NOTE LOADED", noteDetected)`? It seems like evaluating the boolean argument again isn't necessary.